middle_end/flambda2/simplify_shared/closure_offsets.ml

```diff
@@ -616,34 +679,135 @@ module Greedy = struct
   let assign_closure_offsets state env =
     fold_on_unallocated_closure_slots assign_slot_offset env state
 
-  let assign_env_var_offsets state env =
-    fold_on_unallocated_env_var_slots assign_slot_offset env state
-
-  (* Tansform an internal accumulator state for slots into an actual mapping
-     that assigns offsets.*)
-  let finalize state =
-    let env = state.imported_offsets in
-    let env = assign_closure_offsets state env in
-    let env = assign_env_var_offsets state env in
-    env
+  let assign_env_var_offsets ~used_closure_vars state env =
+    fold_on_unallocated_env_var_slots ~used_closure_vars assign_slot_offset env
+      state
+
+  (* Ensure closure ids/env vars that are used in projections in the current
+     compilation unit are present in the offsets returned by finalize *)
+  (* CR mshinwell: these [used_closure_ids] and [used_closure_vars] sets contain
+     more than just the ones used in projections *)
```

Comment:
This CR should have been addressed by the latest changes to `Name_occurrences` (only projections are recorded now)

Reply:
I removed the CR